ts/modules/broken-axis.src.ts

```diff
@@ -484,57 +485,48 @@ class BrokenAxis {
                 eventName: string,
                 y: (number|null|undefined);
 
-            if (!axis) {
-                return; // #5950
-            }
-
-            const brokenAxis = axis.brokenAxis;
-
-            keys.forEach(function (key: string): void {
-                breaks = brokenAxis && brokenAxis.breakArray || [];
-                threshold = axis.isXAxis ?
-                    axis.min :
-                    pick(series.options.threshold, axis.min);
-                points.forEach(function (point: Point): void {
-                    y = pick(
-                        (point as any)['stack' + key.toUpperCase()],
-                        (point as any)[key]
-                    );
-                    breaks.forEach(function (brk: AxisBreakObject): void {
-                        eventName = false as any;
-
-                        if (
-                            (
-                                (threshold as any) < brk.from &&
-                                (y as any) > brk.to
-                            ) ||
-                            (
-                                (threshold as any) > brk.from &&
-                                (y as any) < brk.from
-                            )
-                        ) {
-                            eventName = 'pointBreak';
-
-                        } else if (
-                            (
-                                (threshold as any) < brk.from &&
-                                (y as any) > brk.from &&
-                                (y as any) < brk.to
-                            ) ||
-                            (
-                                (threshold as any) > brk.from &&
-                                (y as any) > brk.to &&
-                                (y as any) < brk.from
-                            )
-                        ) {
-                            eventName = 'pointInBreak';
-                        }
-                        if (eventName) {
-                            fireEvent(axis, eventName, { point: point, brk: brk });
-                        }
+            if (
+                axis && // #5950
+                axis.brokenAxis &&
+                axis.brokenAxis.hasBreaks
+            ) {
+                const brokenAxis = axis.brokenAxis;
+
+                keys.forEach(function (key: string): void {
+                    breaks = brokenAxis && brokenAxis.breakArray || [];
```

Comment:
If we would test agains `brokenAxis` instead of `axis && axis.brokenAxis`, we do not have to test in the loop again.
```ts
const brokenAxis === (
    axis && // #5950
    axis.brokenAxis
); 
if (
    brokenAxis && 
    brokenAxis.hasBreaks
) {
    keys.forEach(function (key: string): void {
         breaks = brokenAxis.breakArray || [];
```

Reply:
Yes I initially did that but then I needed an additional test for `axis` because that's used below. If I understand your comment correctly...

Reply:
In that case, maybe something like this:
```ts
if (!axis) { // #5950
    return;
}

const brokenAxis = axis.brokenAxis;

if (!brokenAxis || !brokenAxis.hasBreaks) {
    return;
}

keys.forEach(function (key: string): void {
         breaks = brokenAxis.breakArray || [];
```

Reply:
IMHO the existing logic looks cleaner...

Reply:
Okay. I had just code optimization in mind.